package/MDAnalysis/coordinates/RDKit.py

```diff
@@ -95,10 +134,591 @@ def __init__(self, filename, **kwargs):
+    attribute to be present in the topology, else it will fail.
+    It also requires the `bonds` attribute, although they will be automatically
+    guessed if not present.
+    If both `tempfactors` and `bfactors` attributes are present, the conversion
+    will fail, since only one of these should be present.
+    Hydrogens should be explicit in the topology file. If this is not the case,
+    use the parameter `NoImplicit=False` when using the converter to allow
+    implicit hydrogens and disable inferring bond orders and charges.
+
+
+    .. versionadded:: 2.0.0
+
+    .. _RDKit: https://www.rdkit.org/docs/source/rdkit.Chem.rdchem.html#rdkit.Chem.rdchem.Mol
+    """
+
+    lib = 'RDKIT'
+    units = {'time': None, 'length': 'Angstrom'}
+    _cache = dict()
+
+    def convert(self, obj, cache=True, **kwargs):
+        """Write selection at current trajectory frame to
+        :class:`rdkit.Chem.rdchem.Mol`.
+
+        Parameters
+        -----------
+        obj : AtomGroup or Universe
+
+        cache : bool
+            Use a cached copy of the molecule's topology when available. To be
+            used, the cached molecule and the new one have to be made from the
+            same AtomGroup object (same id) and with the same arguments passed
+            to the converter (with the exception of this `cache` argument)
+
+        NoImplicit : bool
+            Prevent adding hydrogens to the molecule (default: True)
+        """
+        try:
+            from rdkit import Chem
+        except ImportError:
+            raise ImportError("RDKit is required for the RDKitConverter but "
+                              "it's not installed. Try installing it with \n"
+                              "conda install -c conda-forge rdkit")
+        try:
+            # make sure to use atoms (Issue 46)
+            ag = obj.atoms
+        except AttributeError:
+            raise TypeError("No `atoms` attribute in object of type {}, "
+                            "please use a valid AtomGroup or Universe".format(
+                                type(obj))) from None
+
+        if cache:
+            # key used to search the cache
+            key = f"<{id(ag):#x}>" + ",".join(f"{key}={value}"
+                                            for key, value in kwargs.items())
+            try:
+                mol = self._cache[key]
+            except KeyError:
+                # only keep the current molecule in cache
+                self._cache.clear()
+                # create the topology
+                self._cache[key] = mol = self.atomgroup_to_mol(ag, **kwargs)
+            # continue on copy of the cached molecule
+            mol = copy.deepcopy(mol)
+        else:
+            self._cache.clear()
+            mol = self.atomgroup_to_mol(ag, **kwargs)
+
+        # add a conformer for the current Timestep
+        if hasattr(ag, "positions") and not np.isnan(ag.positions).any():
```

Comment:
It might we worth throwing a warning here in case someone doesn't realise they have NaNs in their data and is expecting to have coordinates assigned